app/models/project_media.rb

```diff
@@ -518,7 +518,7 @@ def log_relevant_results(klass, id, author_id, actor_session_id)
     unless data[type].blank?
       user_action = data[type].include?(article.id) ? 'relevant_articles' : 'article_search'
       tbi = Bot::Alegre.get_alegre_tbi(self.team_id)
-      similarity_settings = tbi.settings
+      similarity_settings = tbi&.settings || {}
```

Comment:
@melsawy, you can also do `tbi&.settings&.to_h` (please confirm) if you need to save complexity.

Reply:
Done